sdk/core/Azure.Core.Experimental/src/DynamicContent.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Represents the <see cref="DynamicJson"/> sent as part of the Azure.Core.Request.
+    /// </summary>
+    public class DynamicContent : RequestContent
+    {
+        private readonly DynamicJson _body;
+
+        internal DynamicContent(DynamicJson body)
+        {
+            _body = body;
+        }
+
+        internal static RequestContent Create(DynamicJson body) => new DynamicContent(body);
+
+        /// <inheritdoc />
+        public override async Task WriteToAsync(Stream stream, CancellationToken cancellation)
+        {
+            using Utf8JsonWriter writer = new Utf8JsonWriter(stream);
+            _body.WriteTo(writer);
+            await writer.FlushAsync(cancellation).ConfigureAwait(false);
+        }
+
+        /// <inheritdoc />
+        public override void WriteTo(Stream stream, CancellationToken cancellation)
+        {
+            using Utf8JsonWriter writer = new Utf8JsonWriter(stream);
+            _body.WriteTo(writer);
+            writer.Flush();
+        }
+
+        /// <inheritdoc />
+        public override bool TryComputeLength(out long length)
+        {
+            using MemoryStream stream = new MemoryStream();
```

Comment:
This is a very expensive way to calculate the length. We can at least try to cache to resulting content.

Reply:
It might be hard because the referenced `DynamicJson` can change.

Reply:
Yeah, this is my unfortunate doing.  While working on a prototype I found that without implementing this we would get chunked encoding for the request and that was causing a problem with a service I was testing.  It is hard with DynamicJson being mutable.

How often is `TryComputeLength` called on average per request?  I would have expected just once if it returns `true`?

Reply:
>How often is TryComputeLength called on average per request?

once